### ISSUE TITLE
Add `regularization` parameter to `Conv2D` and `Conv2DLayer`

### ISSUE THREAD

**Issue by WZMIAOMIAO** [tensorlayer/tensorlayer#590]
### Issue Description

I want to use regularizer in `Conv2d`, but I find kernel and bias regularizer are `None` in  class `Conv2d` code. Although there are some information in loss fuction, that using `network_all_params` method, I think It's so complicated. How to use regularizer in `Conv2d`, and to get expediently regularizer loss through `tf.get_collection(tf.Graphkeys.REGULARIZATON_LOSSES)` function

### Code in Conv2d
```python
conv2d = tf.layers.Conv2d(
                # inputs=self.inputs,
                filters=n_filter,
                kernel_size=filter_size,
                strides=strides,
                padding=padding,
                data_format='channels_last',
                dilation_rate=dilation_rate,
                activation=act,
                use_bias=(False if b_init is None else True),
                kernel_initializer=W_init,  #None,
                bias_initializer=b_init,  #tf.zeros_initializer(),
                **kernel_regularizer=None,
                bias_regularizer=None,**
                activity_regularizer=None,
                kernel_constraint=None,
                bias_constraint=None,
                trainable=True,
                name=name,
                # reuse=None,
            )
```

**Comment by 2wins** [comment]
@WZMIAOMIAO At present, you have to use `tf.layers.conv2d` if you want to use a regularizer in [regularizer.py](https://github.com/tensorflow/tensorflow/blob/r1.8/tensorflow/contrib/layers/python/layers/regularizers.py).
@DEKHTIARJonathan I think we need to support regularization in some layers as well as `tl.layers.Conv2D`.

**Comment by DEKHTIARJonathan** [comment]
@2wins Do yo think of anything specific missing ?
I will add it to the roadmap ;)

**Comment by 2wins** [comment]
@DEKHTIARJonathan If you look at `tl.layers.Conv2D` and `tl.layers.Conv2DLayer`, there are no arguments for regularizations of kernel and bias.

**Comment by DEKHTIARJonathan** [comment]
OIC ! I'll update the roadmap

**Comment by zsdonghao** [comment]
hi, this is a way to add regularisation.

```python
l2 = 0
for w in tl.layers.get_variables_with_name('W_conv2d', train_only=True, printable=False):
    l2 += tf.contrib.layers.l2_regularizer(1e-4)(w)
cost = tl.cost.cross_entropy(y, y_) + l2
```

**Comment by WZMIAOMIAO** [comment]
Thank you for your help! I know using name scope to add regularization, but I wish developers can add regularization in Conv2 Layer.
